tests/FakeItEasy.Specs/EventRaisingSpecs.cs

```diff
@@ -390,6 +390,102 @@ public void ActionEvent()
+
+            "When I raise the event as freeform specifying an argument of a derived type"
+                .x(() => Fake.ReferenceTypeEvent +=
+                    Raise.FreeForm<ReferenceTypeEventHandler>.With(this.derivedReferenceTypeEventArgs));
+
+            "Then the supplied value is passed as the event argument"
+                .x(() => CapturedArgs1.Should().BeSameAs(this.derivedReferenceTypeEventArgs));
+        }
+
+        [Scenario]
+        public void FreeformReferenceTypeEventHandlerWithInvalidArgumentType(
+            Exception exception)
+        {
+            const string ExpectedMessage =
+                "The event has the signature (FakeItEasy.Specs.ReferenceType), " +
+                "but the provided arguments have types (System.Collections.Hashtable).";
+
+            "Given an event of a custom delegate type taking only a reference type as an argument"
+                .See(() => nameof(Fake.ReferenceTypeEvent));
+
+            "When I raise the event as freeform specifying an argument of an invalid type"
+                .x(() => exception = Record.Exception(() =>
+                        Fake.ReferenceTypeEvent += Raise.FreeForm<ReferenceTypeEventHandler>.With(new Hashtable())));
+
+            "Then the call fails with a meaningful message"
+                .x(() => exception.Should().BeAnExceptionOfType<FakeConfigurationException>().And
+                    .Message.Should().Be(ExpectedMessage));
+        }
+
+        [Scenario]
+        public void FreeformValueTypeEventHandlerWithNullValue(
+            Exception exception)
+        {
+            "Given an event of a custom delegate type taking only a value type as an argument"
+                .See(() => nameof(Fake.ValueTypeEvent));
+
+            "When I raise the event as freeform specifying a null argument"
+                .x(() => exception = Record.Exception(() =>
+                        Fake.ValueTypeEvent += Raise.FreeForm<ValueTypeEventHandler>.With((object)null)));
+
+            "Then the call fails with a meaningful message"
+                .x(() => exception.Should().BeAnExceptionOfType<FakeConfigurationException>()
+                    .And.Message.Should().Be(
+                        "The event has the signature (System.Int32), but the provided arguments have types (NULL)."));
+        }
+
+        [Scenario]
+        public void FreeformActionEvent()
+        {
+            "Given an event of type Action"
+                .See(() => nameof(Fake.ActionEvent));
+
+            "When I raise the event as freeform specifying the arguments"
+                .x(() => Fake.ActionEvent += Raise.FreeForm<Action<int, bool>>.With(19, true));
+
+            "Then the first value is passed as the first event argument"
+                .x(() => CapturedArgs1.Should().Be(19));
+
+            "Then the second value is passed as the second event argument"
+                .x(() => CapturedArgs2.Should().Be(true));
+        }
+

```

Comment:
I didn't look closely, but I assume these scenarios are equivalent to the dynamic free-form ones?

Reply:
Yup. Copy, paste, and the minimum modification to make them apply to the new method.